crates/services/p2p/src/heartbeat/handler.rs

```diff
@@ -192,12 +148,13 @@ impl ConnectionHandler for HeartbeatHandler {
         }
 
         loop {
-            if self.failure_count >= self.config.max_failures.into() {
-                // Request from `Swarm` to close the faulty connection
-                return Poll::Ready(ConnectionHandlerEvent::Close(
-                    HeartbeatFailure::Timeout,
-                ))
-            }
+            // TODO: Close connection properly: https://github.com/libp2p/rust-libp2p/pull/4755
+            // if self.failure_count >= self.config.max_failures.into() {
+            //     // Request from `Swarm` to close the faulty connection
+            //     return Poll::Ready(ConnectionHandlerEvent::Close(
+            //         HeartbeatFailure::Timeout,
+            //     ))
+            // }
```

Comment:
Do we need this?

Reply:
Hmm. As it currently works, an unresponsive peer's reputation is going to decay naturally over time. If it runs out it will be banned, not just disconnected.

Would we prefer an unresponsive peer not be banned, like you're saying?

Reply:
Decay doesn't increase the negative score of the peer. It only decreases the value. So it can't cause a disconnection from the peer.

Reply:
I guess you're right since `MIN_APP_SCORE` is negative.

Reply:
> I think it is still better to close/restart the connection if the peer returns nothing to us for a while. But not block him=)

You mean you like this behavior?

The problem, and sorry I didn't clarify this in the original comment, is that there is no longer a `Close` variant of the `ConnectionHandlerEvent`. So I commented it out. I'll create an issue as this behavior isn't captured in any tests.

https://github.com/FuelLabs/fuel-core/issues/1567